nightmare_firmware/utils.h

```diff
@@ -1,7 +1,24 @@
 #ifndef UTILS_H
 #define UTILS_H
 
-float fmap(float x, float in_min, float in_max, float out_min, float out_max); //floating point proportion
-float limit(float val, float downlimit, float uplimit);                        //cut variable if limits exceded
+/**
+ * @brief  linear interpolation in_min:in_max=out_min:out_max
+ * @param  x: value relative to in_min..in_max range
+ * @param  in_min: lower input reference for interpolation
+ * @param  in_max: higher input reference for interpolation
+ * @param  out_min: lower out reference for interpolation
+ * @param  out_max: higher out reference for interpolation
+ * @retval value relative to out_min..out_max 
+ */
+float map(float x, float in_min, float in_max, float out_min, float out_max);
+
+/**
+ * @brief  clamp value to range
+ * @param  val: input value
+ * @param  downlimit: minimum value
+ * @param  uplimit: maximum value
+ * @retval clamped value
+ */
+float clamp(float val, float downlimit, float uplimit);
```

Comment:
A better name for `downlimit/uplimit` might be `min/max`, if you want

Reply:
I'll fix right now